roles/cluster_ensure_machineset/tasks/derive_machineset.yml

```diff
@@ -6,9 +6,22 @@
   register: oc_get_machinesets
   failed_when: not oc_get_machinesets.stdout
 
+- name: Check if base_machineset specified
+  set_fact:
+    base_machineset={{ base_machineset | default(oc_get_machinesets.stdout_lines[0]) }}
+
+- name: Ensure that the specified base machineset exists
+  fail: msg="Specified base machineset ( {{ base_machineset }} ) is invalid or does not exist"
+  when: base_machineset not in oc_get_machinesets.stdout_lines
+
+- name: Ensure that the base machineset exists
+  command: oc get machinesets/{{ base_machineset }} -oname
```

Comment:
you need to specify the namespace here `-n openshift-machine-api`, otherwise it will only work when you're in the `openshift-machine-api` namespace